api/lib/infrastructure/repositories/campaign-collective-result-repository.js

```diff
@@ -108,24 +79,18 @@ function _computeKERankWithinUserAndSkillByDateDescending(qb) {
   });
 }
 
-function _groupByCompetenceId(targetedSkills) {
-  const targetedSkillsByCompetenceId = _.groupBy(targetedSkills, 'competenceId');
-
-  return { targetedSkillsByCompetenceId };
-}
-
-function _forgeCampaignCompetenceCollectiveResults(campaignId, competences, participantCount, targetedSkillsByCompetenceId, participantsKECountByCompetenceId) {
-  return _(targetedSkillsByCompetenceId).keys().map((competenceId) => {
-    const competence = _.find(competences, { id: competenceId });
-    const averageValidatedSkills = (participantCount && competenceId in participantsKECountByCompetenceId) ?
-      participantsKECountByCompetenceId[competenceId] / participantCount : 0;
+function _forgeCampaignCompetenceCollectiveResults(campaignId, targetProfile, participantCount, participantsKECountByCompetenceId) {
+  return _(targetProfile.competences).map((competence) => {
+    const averageValidatedSkills = (participantCount && competence.id in participantsKECountByCompetenceId) ?
+      participantsKECountByCompetenceId[competence.id] / participantCount : 0;
```

Comment:
Je trouve cette ternaire peu lisible, je propose le refacto suivant : 

```suggestion
    let averageValidatedSkills = 0;
    if (participantCount && competence.id in participantsKECountByCompetenceId) {
      averageValidatedSkills = participantsKECountByCompetenceId[competence.id] / participantCount;
    }
```